atom/browser/api/atom_api_top_level_window.cc

```diff
@@ -952,6 +952,17 @@ void TopLevelWindow::RemoveFromParentChildWindows() {
   parent->child_windows_.Remove(weak_map_id());
 }
 
+void TopLevelWindow::EmitEvent(base::StringPiece eventName) {
+  Emit(eventName);
+}
+
+void TopLevelWindow::EmitEventSoon(base::StringPiece eventName) {
```

Comment:
Question: Can we avoid the `EmitEvent` method and make this more generic by putting `EmitSoon` onto the `EventEmitter` base class?

Reply:
Right now the function is slightly brittle because it requires exactly one argument. It would be preferable if that could be resolved before this is opened up for a wider scope in the code.

The issue I mentioned above is the cause for this: `EmitEvent()` exists solely to make the compiler happy because I didn't see how to use the existing `EventEmitter::Emit()`.

To be more specific, this was the compiler error:

```text
      base::BindOnce(&TopLevelWindow::Emit, weak_factory_.GetWeakPtr(), "blur"));
      ^~~~~~~~~~~~~~
../../base/bind.h:179:1: note: candidate template ignored: couldn't infer template argument 'Functor'
```

Reply:
```
base::BindOnce(&TopLevelWindow::Emit<base::StringPiece>, weak_factory_.GetWeakPtr(), "blur"));
```

My understanding of that error is you need to do ☝️

Reply:
Aha! I didn't know about `base::IgnoreResult()` and was unsure how to get rid of the bool return value from Emit, since weak ptrs can only be bound to functions returning void. Thanks for the pointer :+1: